app/src/test/java/com/codesoom/assignment/application/TaskServiceTest.java

```diff
@@ -0,0 +1,92 @@
+package com.codesoom.assignment.application;
+
+import com.codesoom.assignment.TaskNotFoundException;
+import com.codesoom.assignment.models.Task;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.util.List;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+class TaskServiceTest {
+    // 1. list -> getTasks
+    // 2. detail -> getTask (with ID)
+    // 3. create -> createTask (with source)
+    // 4. update -> updateTask (with ID, source)
+    // 5. delete -> deleteTask (with ID)
+
+    private TaskService taskService;
+
+    private static final String TASK_TITLE = "test";
+    private static final String UPDATE_POSTFIX = "!!!";
+
+    @BeforeEach
+    void setUp() {
+        taskService = new TaskService();
+
+        Task task = new Task();
+        task.setTitle(TASK_TITLE);
+
+        taskService.createTask(task);
+    }
+
+    @Test
+    void getTasks() {
+        List<Task> tasks = taskService.getTasks();
+
+        assertThat(tasks).hasSize(1);
+
+        Task task = tasks.get(0);
+        assertThat(taskService.getTasks()).hasSize(1);
+    }
+
+    @Test
+    void getTaskWithValidId() {
+        Task task = taskService.getTask(1L);
+        assertThat(task.getTitle()).isEqualTo(TASK_TITLE);
+    }
+
+    @Test
+    void getTaskWithInvalideId() {
+        assertThatThrownBy(() -> taskService.getTask(100L))
+                .isInstanceOf(TaskNotFoundException.class);
+    }
+
+    @Test
+    void createTask() {
+        int oldSize = taskService.getTasks().size();
+
+        Task task = new Task();
+        task.setTitle(TASK_TITLE);
+
+        taskService.createTask(task);
+
+        int newSize = taskService.getTasks().size();
+
+        assertThat(newSize - oldSize).isEqualTo(1);
+    }
+
+    @Test
+    void updateTask() {
+        Task source = new Task();
+        source.setTitle(TASK_TITLE + UPDATE_POSTFIX);
+
+        taskService.updateTask(1L, source);
+
+        Task task = taskService.getTask(1L);
+        assertThat(task.getTitle()).isEqualTo(TASK_TITLE + UPDATE_POSTFIX);
+    }
+
+    @Test
+    void deleteTask() {
+        int oldSize = taskService.getTasks().size();
+
+        taskService.deleteTask(1L);
+
+        int newSize = taskService.getTasks().size();
+
+        assertThat(oldSize - newSize).isEqualTo(1);
```

Comment:
```suggestion
        assertThat(newSize - oldSize).isEqualTo(-1);
```

이렇게 해서 감소한다는 것을 표현하는 것은 너무 과할까요?